### ISSUE TITLE
New function `isEmpty`

### ISSUE THREAD

**Issue by Alexnortung** [uhyo/ts-array-length#6]
Let's say I want to get the first item in an array, but i want to make sure that the array is empty first. you can do it in these two ways

```ts
if (isNonEmpty(arr)) {
  const first = arr[0];
}
```

or

```ts
if (!isNonEmpty(arr)) {
  return;
}
const first = arr[0];
```

I would normally prefer the second option as it would allow for less nesting.

The problem is that `!isNonEmpty(arr)` doesn't really look good and is kinda weird to read (not is not empty). Here it would be much easier to just use `isEmpty(arr)`. This would also read nicely if you want to use `!isEmpty(arr)` and still feels intuitive.

**Comment by uhyo** [comment]
You're right that `isEmpty` is more natural and erogonomic way of expressing this functionality. 😄

However, TypeScript doesn't allow us to do this. Fow now, this must be `isNonEmpty` in order to perform proper type narrowing.

[Playground](https://www.typescriptlang.org/play/?noUncheckedIndexedAccess=true#code/GYVwdgxgLglg9mABDAzgUQLYAcoE8A8AKgHwAUAhgE6Xm4BcilApuQCYIA2uihA2gLoBKBlRrdUjFuzBdEAxAG8AUIklQQlJKNoA6DkzABzKAAtEAXkuIADAG4lAXyVLQkWAkQgUTAILVaFP70iGAgGABGTJQCgooqyMCIpKiYOLiBYoKxyqqqzOqa9qpOqhAIKFCIwDCUFRaI2ri81vxFiAD07bmIAHoA-PGdiHBwWCgANIgABtW1UFPIKIgVMBwc06ERUYgAPp5grEzVYEysC6QA7jCmIXAAqpAmTBAA1qcAkgdMAB6nPhAQJgoFCCRzOVzQeBIVAAOQQqTwRDIjQYzDYnG4fCEIiCi0k6Jk3F4hEmOjJWP4cTyTAKWiCegMxjMAEIrHYwS5wJCPF5fEEAEwZWgMTaRaJCKkJJLM2Hw7B4IW4LKS6m0tolRBlMB1WZ1cwNILNVqDLq5fomxAXOAXSYzGoVBYSKaiqILJ7MRxAA)

```ts
function isEmpty<T>(array: readonly T[]): array is readonly [] {
  return array.length === 0;
}

function useArray(array: number[]) {
  if (isEmpty(array)) {
    return;
  }
  const first = array[0];
  //    ^?
  // oops, `first` is still `number | undefined` (with noUncheckedIndexedAccess)
}

function isNonEmpty<T>(array: readonly T[]): array is readonly [T, ...T[]] {
  return array.length !== 0;
}

function useArray2(array: number[]) {
  if (!isNonEmpty(array)) {
    return;
  }
  const first = array[0];
  //    ^?
  // wow, `first` is `number` here
}
```

**Comment by Alexnortung** [comment]
Ahh I see, that is quite annoying, but it makes sense :)
